Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3928,6 +3928,10 @@ GstElement* MediaPlayerPrivateGStreamer::createVideoSink()
         g_signal_connect_swapped(m_videoSink.get(), "repaint-cancelled", G_CALLBACK(repaintCancelledCallback), this);
     }
 
+    // In some context, we could end up with a/v desync so we need to ensure late frames are discarded.
+    uint64_t maxLateness = 100 * GST_MSECOND;
+    g_object_set(m_videoSink.get(), "max-lateness", maxLateness, nullptr);
```

Comment:
is `m_videoSink` a basesink sub-class in all cases? At least our custom GL sink isn't, it's a bin, so setting max-lateness in that situation will cause havoc :)

Reply:
If the property does not exist, it should be harmless other than maybe getting a warning. I see two options here:
1. Adding the property to all individual sinks.
2. Adding the property here but checking for special sinks like the GL one.

Fun fact: I had the code just for the GL sink and decided to move it there to make it more general :sweat_smile:

Reply:
Well yes, since no check is done in this current patch, i don't see how any issue would be solved unless you're lucky at runtime and you get a basesink sub-class.

So can you check all the cases we support (GL, non-GL, punch-hole) and also don't do this if the source is a MediaStream please, because sync is disabled for that case anyway.

Reply:
I think I solved this.